### ISSUE TITLE
Heading Issues in Catalog, Book View, and Reader

### ISSUE THREAD

**Issue by avaaec** [pressbooks/pressbooks#1788]
### Description

This is a collection of all of the heading issues found in the outward facing interfaces for pressbooks.

### Steps to Reproduce

1. Visit https://uw.pressbooks.pub/catalog/ and explore the headings
2. Visit https://uw.pressbooks.pub/lafrancesauvee/ and explore the headings
3. Visit [a section of the book, preferably with lots of interesting content](https://uw.pressbooks.pub/lafrancesauvee/chapter/the-princess-of-lamballe-marie-louise-therese-of-savoy-carignan-superintendent-of-the-queens-household/)

**Problems** 
1. In the catalog page the filter controls are h2, they do not need to be headings because they are immediately after the "Catalog" h1 so they are easy to discover and navigate to for assistive tech users.
2. On the book page (pressbooks.pub/lafrancesauvee/) the metadata section does not have the same h3s as the other sections. This seems like a pretty simple bug, they should be consistent with the other h2 sections of the page.
3. In the reader interface for the book, the "Content" h3 is above the "[title of the section]" h1. Because of this, the "Content" heading is not semantically associated as a child of the h1. This is a problem, it needs to be rendered after the h1 to make sense. Also it should be an h2 not an h3.
4. The content of the book has it's own headings which do not fit into the hierarchy of the page. For example in the book section page linked in steps to reproduce, the book content starts with an h1 and contains a random empty h6 tag in the middle. Regardless of what headings the book content uploaded had, it is Pressbooks responsibility to render the content in a way that does not destroy the heading structure of the page. The h1 in the book content should be a h2 and the random h6 should probably be discarded. Any additional headings in the book content should be assigned heading levels in logical order, h3 and so on.

**Comment by mcgratay** [comment]
Could you elaborate a little on no. 3 if possible? Is it the chapter interface, with "Content" being the table of contents dropdown? A screenshot would be great. Thanks :)

**Comment by SteelWagstaff** [comment]
TODO: Find out why 'Contents' in reading-header TOC is an h3 element.

**Comment by avaaec** [comment]
Hi @mcgratay! Sorry for the late response. Attached is a screen shot of the page with the headings hightlighted using https://accessibility-bookmarklets.org/. You can see that the h3 heading appears above the h1 heading in a linearized left to right top to bottom view. This means that for non visual users or people who use alternative input methods the headings are effectively out of order. This is not a very simple problem to fix, likely one of the two headings will need to visually move on the page so that the headings make semantic sense.
![Screenshot from 2019-11-07 13-05-20](https://user-images.githubusercontent.com/51925015/68427903-e264d000-015f-11ea-875b-dc0eeb7d4a92.png)

**Comment by avaaec** [comment]
all of these headings problems are UW ATS priority 1

**Comment by SteelWagstaff** [comment]
For the in-book ToC, we can fix by changing the 'Contents' ToC title from an `<h3>` to a `<div>` element: https://github.com/pressbooks/pressbooks-book/blob/108f5d4d0324adc5436a16cd2edac0f70646269a/header.php#L106

We will also need to fix the javascript that modifies this `<h3>` to receive a button: https://github.com/pressbooks/pressbooks-book/blob/1c687906d47b278ddc3d263fda712b869632277c/assets/src/scripts/routes/common.js#L89-L103 and fix the relevant SCSS that targets the `<h3>` that has this class: https://github.com/pressbooks/pressbooks-book/blob/df90c939b473d4a4e37f3d6cc7e8304a3d5009c7/assets/src/styles/layouts/_header.scss#L93

**Comment by connermwallace** [comment]
University of Washington Accessibility Technology Services has inspected this issue and we have verified that it has been fixed. Version: 5.3.2.
